packages/back-end/src/routers/sdk-connection/sdk-connection.controller.ts

```diff
@@ -0,0 +1,144 @@
+import type { Response } from "express";
+import { AuthRequest } from "../../types/AuthRequest";
+import { getOrgFromReq } from "../../services/organizations";
+import {
+  SDKConnectionInterface,
+  CreateSDKConnectionParams,
+  EditSDKConnectionParams,
+  ProxyTestResult,
+} from "../../../types/sdk-connection";
+import {
+  createSDKConnection,
+  deleteSDKConnectionById,
+  editSDKConnection,
+  findSDKConnectionById,
+  findSDKConnectionsByOrganization,
+  testProxyConnection,
+} from "../../models/SdkConnectionModel";
+
+export const getSDKConnections = async (
+  req: AuthRequest,
+  res: Response<{
+    status: 200;
+    connections: SDKConnectionInterface[];
+  }>
+) => {
+  const { org } = getOrgFromReq(req);
+  const connections = await findSDKConnectionsByOrganization(org.id);
+  res.status(200).json({
+    status: 200,
+    connections,
+  });
+};
+
+export const postSDKConnection = async (
+  req: AuthRequest<Omit<CreateSDKConnectionParams, "organization">>,
+  res: Response<{
+    status: 200;
+    connection: SDKConnectionInterface;
+  }>
+) => {
+  const { org } = getOrgFromReq(req);
+  const params = req.body;
+
+  req.checkPermissions("manageEnvironments", params.project, [
+    params.environment,
+  ]);
+
+  const doc = await createSDKConnection({ ...params, organization: org.id });
```

Comment:
I notice we aren't using the `validateRequestMiddleware` and we aren't doing any kind of server-side validation here, which means the client-side can post any params and we are including them all without checking. The code generation I set up sets us up with object validation with zod. We can do `yarn plop` and choose "router." For existing routers, we can look at existing routers for reference. Here's [an example](https://github.com/growthbook/growthbook/blob/main/packages/back-end/src/routers/event-webhooks/event-webhooks.router.ts). Similarly for the `putSDKConnection` as well. Get and Delete should be ok since it's just the path params and Express handles that at the routing layer.

Reply:
I moved the validation logic to the model class for create and update.

Reply:
This will be caught by the global error handler and return 401 instead of a 400, right?